src/Compilers/CSharp/Portable/BoundTree/BoundNodeExtensions.cs

```diff
@@ -66,5 +66,31 @@ public static bool IsConstructorInitializer(this BoundCall call)
             node.WasCompilerGenerated = true;
             return node;
         }
+
+        public static bool ContainsAwaitExpression(this ImmutableArray<BoundExpression> expressions)
+        {
+            var visitor = new ContainsAwaitVisitor();
+            foreach (var expression in expressions)
+            {
+                visitor.Visit(expression);
+                if (visitor.ContainsAwait)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private class ContainsAwaitVisitor : BoundTreeWalkerWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
+        {
+            public bool ContainsAwait = false;
+
+            public override BoundNode? VisitAwaitExpression(BoundAwaitExpression node)
```

Comment:
>VisitAwaitExpression [](start = 39, length = 20)

Consider also overriding non-strongly type Visit to short-circuit if we already found an await.